# Release 3.0.0

golluroSICKAG opened the pull request:
# Version 3.0.0

## New features
- Supports FlowConfig feature to set images to process / provide results
- Provide version of module via 'OnNewStatusModuleVersion'
- Function 'getParameters' to provide PersistentData parameters
- Check if features of module can be used on device and provide this via 'OnNewStatusModuleIsActive' event / 'getStatusModuleActive' function
- Function to 'resetModule' to default setup

## Improvements
- New UI design available (e.g. selectable via CSK_Module_PersistentData v4.1.0 or higher), see 'OnNewStatusCSKStyle'
- check if instance exists if selected
- 'loadParameters' returns its success
- 'sendParameters' can control if sent data should be saved directly by CSK_Module_PersistentData
- Changed log level of some messages from 'info' to 'fine'
- Added UI icon and browser tab information

## Bugfix
- Never deregistered from events

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

golluroSICKAG committed:
# Release 3.0.0

## New features
- Supports FlowConfig feature to set images to process / provide results
- Provide version of module via 'OnNewStatusModuleVersion'
- Function 'getParameters' to provide PersistentData parameters
- Check if features of module can be used on device and provide this via 'OnNewStatusModuleIsActive' event / 'getStatusModuleActive' function
- Function to 'resetModule' to default setup

## Improvements
- New UI design available (e.g. selectable via CSK_Module_PersistentData v4.1.0 or higher), see 'OnNewStatusCSKStyle'
- check if instance exists if selected
- 'loadParameters' returns its success
- 'sendParameters' can control if sent data should be saved directly by CSK_Module_PersistentData
- Changed log level of some messages from 'info' to 'fine'
- Added UI icon and browser tab information

## Bugfix
- Never deregistered from events

golluroSICKAG committed:
- Update of UI settings

tarnamaSICKAG reviewed (commented):
Bei "Registered on image event" fehlt der Beispieltext ("CSK_ImagePlayer.OnNewImage)

Bei "Use Masking" active in Step 4, kleiner bug:
Die angezeigte ROI wird nicht von rectangle auf circle umgestellt wenn:
- Circle als "mask type" ausgewählt ist
- dann Reset Mask geklickt wird
- dann wieder auf circle gegangen wird
Danach muss man als Workaround "Mask Type" auf rectangle stellen und anschließend wieder auf circle.
Rest sieht gut aus :)

golluroSICKAG committed:
- Fixed issues of review

Bugfix
- Error if module is not active but 'getInstancesAmount' was called
- processInstanceNUM did not work after deregistering from event to process images

golluroSICKAG commented:
Issues sould be fixed now.
Thx for the review.

golluroSICKAG committed:
- Removing navigation bar entries
